src/daemon/mod.rs

```diff
@@ -171,11 +180,12 @@ pub(super) async fn start(
 
     let keystore = Arc::new(RwLock::new(keystore));
 
-    let chain_data_path = chain_path(&config);
-    let db = Arc::new(ManyCar::new(Arc::new(open_proxy_db(
-        db_root(&chain_data_path),
-        config.db_config().clone(),
-    )?)));
+    if opts.detach {
+        unblock_parent_process()?;
+    }
```

Comment:
Detaching before the RPC is available looks wrong. Why did you change this?

Reply:
Moved TCP endpoint binding logic to before calling `unblock_parent_process`